sdk/core/Azure.Core.Experimental/tests/DynamicJsonTests.cs

```diff
@@ -405,6 +405,308 @@ public void DisposingAChildDisposesTheParent()
+        }
+
+        [Test]
+        public void CanSetPascalCaseNoMapping()
+        {
+            string json = """{ "Foo": 1 }""";
+
+            DynamicJsonOptions options = new()
+            {
+                PropertyNameCasing = DynamicJsonNameMapping.None
+            };
+            dynamic dynamicJson = new BinaryData(json).ToDynamic(options);
+
+            // This adds a new property, since it doesn't find `Foo`.
+            dynamicJson.foo = 2;
+
+            // New property access
+            dynamicJson.bar = 3;
+
+            Assert.AreEqual(2, (int)dynamicJson.foo);
+            Assert.AreEqual(1, (int)dynamicJson.Foo);
+            Assert.AreEqual(3, (int)dynamicJson.bar);
+            Assert.AreEqual(null, dynamicJson.Bar);
+
+            // This updates the PascalCase property and not the camelCase one.
+            dynamicJson.Foo = 4;
+
+            // This creates a new PascalCase property.
+            dynamicJson.Bar = 5;
+
+            Assert.AreEqual(2, (int)dynamicJson.foo);
+            Assert.AreEqual(4, (int)dynamicJson.Foo);
+            Assert.AreEqual(3, (int)dynamicJson.bar);
+            Assert.AreEqual(5, (int)dynamicJson.Bar);
+        }
+
+        [Test]
+        public void CanSetPascalCasePascalGetters()
+        {
+            string json = """{ "Foo": 1 }""";
+
+            DynamicJsonOptions options = new()
+            {
+                PropertyNameCasing = DynamicJsonNameMapping.PascalCaseGetters
+            };
+            dynamic dynamicJson = new BinaryData(json).ToDynamic(options);
+
+            // This property doesn't exist, so it creates a new camelCase property.
+            dynamicJson.foo = 2;
+
+            // New property is created as camelCase
+            dynamicJson.bar = 3;
+
+            Assert.AreEqual(2, (int)dynamicJson.foo);
+            Assert.AreEqual(1, (int)dynamicJson.Foo);
+            Assert.AreEqual(3, (int)dynamicJson.bar);
+            Assert.AreEqual(3, (int)dynamicJson.Bar);
+
+            // This updates the PascalCase property and not the camelCase one.
+            dynamicJson.Foo = 4;
+
+            // The PascalCase getter finds `bar`, so it updates the camelCase property.
+            dynamicJson.Bar = 5;
+
+            // New property is created as PascalCase
+            dynamicJson.Baz = 6;
+
+            Assert.AreEqual(2, (int)dynamicJson.foo);
+            Assert.AreEqual(4, (int)dynamicJson.Foo);
+            Assert.AreEqual(5, (int)dynamicJson.bar);
+            Assert.AreEqual(5, (int)dynamicJson.Bar);
+            Assert.AreEqual(null, dynamicJson.baz);
+            Assert.AreEqual(6, (int)dynamicJson.Baz);
+        }
+
+        [Test]
+        public void CanSetPascalCasePascalGettersCamelSetters()
+        {
+            string json = """{ "Foo": 1 }""";
+
+            dynamic dynamicJson = new BinaryData(json).ToDynamic();
+
+            // Existing property access does not add a camelCase property.
```

Comment:
Why wouldn't new properties just match whatever casing the user specifies via the dynamic?

Reply:
It is because Azure REST API guidance is to camelCase JSON, and .NET guidance is to PascalCase properties.  If services follow the stated guidance, everything should work correctly.  If services do not, users will need to override the default behavior.